src/sentry/incidents/endpoints/validators.py

```diff
@@ -103,7 +109,85 @@ def validate(self, attrs):
         And this should be moved to a metric alert specific app, probably `incidents/`
         """
         attrs = super().validate(attrs)
-        conditions = attrs["data_conditions"]
-        if len(conditions) > 2:
-            raise serializers.ValidationError("Too many conditions")
+        # TODO should we limit the number of data sources?
         return attrs
+
+    def update_data_condition(self, instance, data_conditions):
+        """
+        Update the data condition if it already exists, create one if it does not
```

Comment:
Sounds like this should be more like update the data condition if we've been passed an id, create a new one and attach it to the dcg if not?